packages/code-block/partials/code-lines/index.tsx

```diff
@@ -0,0 +1,240 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+/**
+ * Note: lines of code are expected to be stable. If we need to work with
+ * dynamic code blocks in the future, we could assign random unique IDs
+ * to each line during the `linesOfCode` `useMemo` function.
+ *
+ * For now, we disable react/no-array-index key for the entire file.
+ */
+/* eslint-disable react/no-array-index-key */
+
+import React, { PropsWithChildren, ReactElement, useMemo } from 'react'
+import classNames from 'classnames'
+import parseHighlightedLines from '../../utils/parse-highlighted-lines'
+import splitJsxIntoLines from './utils/split-jsx-into-lines'
+import splitHtmlIntoLines from './utils/split-html-into-lines'
+import s from './code-lines.module.css'
+
+interface CodeLinesProps {
+  code: string | ReactElement
+  language?: string
+  lineNumbers?: boolean
+  highlight?: string
+  hasFloatingCopyButton?: boolean
+  wrapCode?: boolean
+}
+
+/**
+ * Render the provided code into separate line elements,
+ * accounting for all provided options.
+ */
+function CodeLines({
+  code,
+  language,
+  lineNumbers,
+  highlight,
+  hasFloatingCopyButton,
+  wrapCode,
+}: CodeLinesProps) {
+  // Parse out an array of integers representing which lines to highlight
+  const highlightedLines = parseHighlightedLines(highlight) as number[]
```

Comment:
Does this need to be cast?

Reply:
For now yes, as otherwise it seems the inferred type of `highlightedLines` ends up being `never[]`, and we get a type error later at `highlightedLines.indexOf(number) !== -1`.

Ideally I'd have jumped in and converted `parseHighlightedLines` to TypeScript as well, but I didn't quite have the energy 😅